beacon-chain/db/kv/state.go

```diff
@@ -196,6 +204,18 @@ func (k *Store) DeleteState(ctx context.Context, blockRoot [32]byte) error {
 			}
 		}
 
+		enc = bkt.Get(blockRoot[:])
+		if enc == nil {
+			return nil
+		}
+		block := &ethpb.SignedBeaconBlock{}
+		if err := decode(enc, block); err != nil {
```

Comment:
Is there any way around this? Do we track an index of block root to slot?

Could you use the state summary, if it's cheaper?

Reply:
Refactored this to `slotByBlockRoot`
https://github.com/prysmaticlabs/prysm/blob/fd4bb0f8fb244d230fa695b98e71d2eb1c18d144/beacon-chain/db/kv/state.go#L294

It handles panic for nil block, and uses state summary when it's possible